### ISSUE TITLE
koala depends on multi_json 1.8.0

### ISSUE THREAD

**Issue by dan-rising** [arsduo/koala#449]
When trying to exchange an access_token I have been encountering this NameError:

irb(main):030:0> Koala::Facebook::OAuth.new(app, secret).exchange_access_token_info(access_token)
NameError: uninitialized constant MultiJson::ParseError

This was coming from:
koala-1.11.1/lib/koala/oauth.rb:320:in `rescue in parse_access_token'

And here's that method (from koala 1.11.1):
```
def parse_access_token(response_text)
  MultiJson.load(response_text)
rescue MultiJson::ParseError
  response_text.split("&").inject({}) do |hash, bit|
    key, value = bit.split("=")
    hash.merge!(key => value)
  end
end
```
In my case I'm encountering the NameError because multi_json did not define MultiJson::ParseError until v1.8.0 and I am stuck on v1.7.9 of MultiJson.

I have worked around this by defining it myself like so:

```
module MultiJson
  ParseError = MultiJson::LoadError
end
```

This is similar to what's in the MultiJson source, as when they introduced ParseError they also defined aliases for backwards-compatibility, as can be seen here: https://github.com/intridea/multi_json/blob/d85d23b6d77be1934d3d4f03c1167d554da32e7d/lib/multi_json/parse_error.rb

It would be great to either make the minimum version dependency explicit in koala (or perhaps even use one of the aliases they've defined).

**Comment by nitinstp23** [comment]
I am also facing this issue.

**Comment by akalra** [comment]
For anyone else that runs into this, I think the version of multi_json that introduces ParseError is actually 1.9.0 not 1.8.0:
https://github.com/intridea/multi_json/commit/4abb98fe3a90b2a7b3d1594515c8a06042b4a27d#diff-e2ff2853759b2841568a761899e11c24

Thanks!

Ankur

**Comment by arsduo** [comment]
Hi everyone, 

This has now been addressed -- I've merged #469, which uses the older LoadError instead of ParseError  for backward compatibility. Koala 2.2 will be coming out soon.

Sorry it took me so long to address this -- going forward, I'll be updating Koala according to a fixed, 3x/year support schedule for Koala to make sure issues and pull requests get addressed in a relatively timely manner.

Best,

Alex
